src/mon/OSDMonitor.cc

```diff
@@ -922,10 +922,12 @@ OSDMonitor::update_pending_pgs(const OSDMap::Incremental& inc)
     pending_creatings.pgs.erase(pg);
   }
   pending_created_pgs.clear();
-  // PAXOS_PGMAP is less than PAXOS_OSDMAP, so PGMonitor::update_from_paxos()
-  // should have prepared the latest pgmap if any
-  mon->pgservice->maybe_add_creating_pgs(creating_pgs.last_scan_epoch,
-					 &pending_creatings);
+  if (!osdmap.test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) {
```

Comment:
no, i don't think it's backwards. we don't need to sync up with pgmap if all osds are luminous. but we need to do so while some osds are still pre-luminous. 

i try to keep `creating_pgs` updated with the bits from pgmap which is ensured to be updated **before** `CEPH_OSDMAP_REQUIRE_LUMINOUS` is set. so we can have a complete `creating_pgs` set when we are switched to `CEPH_OSDMAP_REQUIRE_LUMINOUS`. assuming following upgrade steps:

1. all mon are luminous, and `FEATURE_LUMINOUS` in monmap is set when the last luminous mon joins the quorum. but not all osds are luminous. so we can not set `CEPH_OSDMAP_REQUIRE_LUMINOUS`
1. the leader starts to update `creating_pgs`, but it is not able have the complete set of pgs being created just by consuming the (inc) osdmap after `FEATURE_LUMINOUS` is set. because

   1. there are some pools is still being created when the `FEATURE_LUMINOUS` is set. so the pgs belonging to that pools could be still being created also. and they only exist in pgmap.
   1. there are some osds are still at pre-luminous, so they keep `PGMonitor` (now `PGMapStatService`) updated with `MPGStats`, but they don't send `MOSDPGCreated` to mon.  that's what `trim_creating_pgs()` is for. please note we also guard it with `!osdmap.test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)`  in `OSDMonitor::encode_pending()`. 
1. but once we have `CEPH_OSDMAP_REQUIRE_LUMINOUS` in osdmap, all osds are luminous. we don't need the pgmap for updating creating_pgs anymore, the luminous osds will ensure this.

am i missing anything?